### DISCUSSION TITLE
sqlite package: package scoped db variable instead of injection

### DISCUSSION THREAD

**Discussion by youdzinn** [benbjohnson/wtf#18] [Q&A] [answered]
Hello Ben,

first of all, thanks for the great stuff!

I'm a newbie to Golang (Spring Boot switcher), and I saw many examples including real production applications when `db` connection instance is private package scoped, like
```
db/
  database.go // with var db *sql.db declared
  entity.go // functions or methods use db variable for CRUD ops
```

Tell me please why use dependency injection instead of such kind of private package variable?

**Comment by benbjohnson** [comment, marked as answer]
Hi Eugene,

The downside to using a global `db` variable in your package is that it can make testing problematic. Especially if you want to parallelize tests to speed them up. I generally try to avoid global variables because it makes it more difficult to change code and reuse code over time.

**Reply by youdzinn** [reply]
Got that, thanks for your help.
